catalog/async/catalog-async-data/src/test/groovy/org/codice/ddf/catalog/async/data/LazyProcessResourceImplSpec.groovy

```diff
@@ -348,4 +348,16 @@ class LazyProcessResourceImplSpec extends Specification {
         then:
         thrown(IOException)
     }
+
+    def 'IOException is thrown when getInputStream is called after close' () {
```

Comment:
nice!

❓ Should we add one to account for the case where `getInputStream()` is never called but `getName()` is to ensure that when `close()` is finally called, the input stream will be closed too? Also maybe modify or add new test cases to verify the resource input stream is properly closed in various situations